std/haxe/Http.hx

```diff
@@ -218,33 +219,56 @@ class Http {
 		var question = path.split("?").length <= 1;
 		if (!post && uri != null) path += (if( question ) "?" else "&") + uri;
 
-		var opts = {
+		var opts: Dynamic = {
 			protocol: parsedUrl.protocol,
 			hostname: host,
 			port: port,
 			method: post ? 'POST' : 'GET',
 			path: path,
 			headers: h
 		};
-		function httpResponse (res) {
-			var s = res.statusCode;
-			if (s != null)
-				me.onStatus(s);
-			var body = '';
-			res.on('data', function (d) {
-				body += d;
-			});
-			res.on('end', function (_) {
-				me.responseData = body;
-				me.req = null;
-				if (s != null && s >= 200 && s < 400) {
-					me.onData(body);
-				} else {
-					me.onError("Http Error #"+s);
-				}
-			});
-		}
-		req = secure ? js.node.Https.request(untyped opts, httpResponse) : js.node.Http.request(untyped opts, httpResponse);
+		var httpResponse = if (requestBytes) {
+			opts.encoding = null;
+			function (res) {
+				var s = res.statusCode;
+				if (s != null)
+					me.onStatus(s);
+				var body = [];
+				res.on('data', function (d/*:js.node.Buffer*/) {
+					body.push(d);
+				});
+				res.on('end', function (_) {
+					var bytes = js.node.Buffer.concat(body).hxToBytes();
+					me.responseData = body.toString();
```

Comment:
I am not sure if ```responseData``` should be filled when binary data is requested.
Maybe it could be left ```null```.